src/main/scala/org/deepdive/inference/datastore/SQLInferenceDataStore.scala

```diff
@@ -459,173 +319,89 @@ trait SQLInferenceDataStore extends InferenceDataStore with Logging {
-    assignVariablesIds(schema)
-
+  
+  // assign variable holdout
+  def assignHoldout(schema: Map[String, _ <: VariableDataType], calibrationSettings: CalibrationSettings) {
     // variable holdout table - if user defined, execute once
-    execute(s"""DROP TABLE IF EXISTS ${VariablesHoldoutTable} CASCADE;
-      CREATE TABLE ${VariablesHoldoutTable}(variable_id bigint primary key);
-      """)
+    ds.dropAndCreateTable(VariablesHoldoutTable, "variable_id bigint primary key")
     calibrationSettings.holdoutQuery match {
-      case Some(query) => execute(query)
-      case None =>
-    }
-
-    // variable observation table
-    execute(s"""DROP TABLE IF EXISTS ${VariablesObservationTable} CASCADE;
-      CREATE TABLE ${VariablesObservationTable}(variable_id bigint primary key);
-      """)
-    calibrationSettings.observationQuery match {
-      case Some(query) => execute(query)
-      case None =>
-    }
-
-    // ground variables
-    schema.foreach { case(variable, dataType) =>
-      val Array(relation, column) = variable.split('.')
-
-      val variableDataType = dataType match {
-        case BooleanType => 0
-        case MultinomialType(x) => 1
-        case RealNumberType => 2
-        case RealArrayType(x) => 3
+      case Some(query) => {
+        log.info("Executing user supplied holdout query")
+        execute(query)
       }
-
-      val cardinality = dataType match {
-        case BooleanType => 2
-        case MultinomialType(x) => x.toInt
-        case RealNumberType => 2
-        case RealArrayType(x) => x.toInt
-      }
-
-      if(variableDataType == 2 || variableDataType == 3){
-
-      } else {
-        // This cannot be parsed in def randFunc for now.
-        // assign holdout - if not user-defined, randomly select from evidence variables of each variable table
-        calibrationSettings.holdoutQuery match {
-          case Some(s) =>
-          case None => execute(s"""
+      case None => {
+        log.info("There is no holdout query, will randomly generate holdout set")
+         // randomly assign variable holdout
+        schema.foreach { case(variable, dataType) =>
+          val Array(relation, column) = variable.split('.')
+          // This cannot be parsed in def randFunc for now.
+          // assign holdout - randomly select from evidence variables of each variable table
+          execute(s"""
             INSERT INTO ${VariablesHoldoutTable}
             SELECT id FROM ${relation}
             WHERE ${randomFunction} < ${calibrationSettings.holdoutFraction} AND ${column} IS NOT NULL;
             """)
         }
       }
+    }
 
+    // variable observation table
+    ds.dropAndCreateTable(VariablesObservationTable, "variable_id bigint primary key")
+    calibrationSettings.observationQuery match {
+      case Some(query) => {
+        log.info("Executing user supplied observation query")  
+        execute(query)
+      }
+      case None => {
+        log.info("There is no o query")
```

Comment:
I am not entirely sure what an observation query is, so I don't know if failing to run it is catastrophic or not.  Is this for debugging, or for indicating the evidence variables?